container/go/pkg/compat/image_index.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2015 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//////////////////////////////////////////////////////////////////////
+package compat
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/types"
+)
+
+type imageIndex struct {
+	platforms   []v1.Platform
+	images      []v1.Image
+	manifest    *v1.IndexManifest
+	rawManifest []byte
+	digest      v1.Hash
+}
+
+// Ensure imageIndex implements the v1.ImageIndex interface.
+var _ v1.ImageIndex = (*imageIndex)(nil)
+
+// NewImageIndex creates a new image index from the given list of platforms and images.
+func NewImageIndex(platforms []v1.Platform, images []v1.Image) (v1.ImageIndex, error) {
+	if len(platforms) != len(images) {
+		return nil, fmt.Errorf("list of platforms and images must have the same number of entries")
+	}
+
+	ii := imageIndex{
+		platforms: platforms,
+		images:    images,
+	}
+
+	if err := ii.init(); err != nil {
+		return nil, err
+	}
+
+	return &ii, nil
+}
+
+func (ii *imageIndex) init() error {
```

Comment:
Rather than an init func it might be more idiomatic to keep this logic within `NewImageIndex`?